Source/Core/Core/HW/WiimoteReal/IOLinux.cpp

```diff
@@ -137,28 +137,43 @@ bool WiimoteLinux::ConnectInternal()
   addr.l2_family = AF_BLUETOOTH;
   addr.l2_bdaddr = m_bdaddr;
   addr.l2_cid = 0;
+  int retry;
 
   // Output channel
   addr.l2_psm = htobs(WM_OUTPUT_CHANNEL);
-  if ((m_cmd_sock = socket(AF_BLUETOOTH, SOCK_SEQPACKET, BTPROTO_L2CAP)) == -1 ||
-      connect(m_cmd_sock, (sockaddr*)&addr, sizeof(addr)) < 0)
+  retry = 0;
+  while ((m_cmd_sock = socket(AF_BLUETOOTH, SOCK_SEQPACKET, BTPROTO_L2CAP)) == -1 ||
+        connect(m_cmd_sock, (sockaddr*)&addr, sizeof(addr)) < 0)
   {
-    WARN_LOG(WIIMOTE, "Unable to open output socket to Wiimote: %s", strerror(errno));
-    close(m_cmd_sock);
-    m_cmd_sock = -1;
-    return false;
+    // If opening channel fails sleep and try again
+    if (retry == 3)
+    {
+      WARN_LOG(WIIMOTE, "Unable to open output socket to Wiimote: %s", strerror(errno));
+      close(m_cmd_sock);
+      m_cmd_sock = -1;
+      return false;
+    }
+    retry++;
```

Comment:
This leaks file descriptors by not close()ing the socket on the first 2 attempts.